src/shared/tools.ts

```diff
@@ -267,3 +269,85 @@ export interface DiffStrategy {
 
 	getProgressStatus?(toolUse: ToolUse, result?: any): ToolProgressStatus
 }
+
+/**
+ * Safely reads a file from the .roo directory with the given filename.
+ * Returns an empty string if the file doesn't exist or cannot be read.
+ *
+ * @param cwd - The working directory
+ * @param file - The filename to read from the .roo directory
+ * @returns A promise that resolves to the file content or empty string
+ */
+export async function readToolOverride(cwd: string, file: string): Promise<string> {
```

Comment:
you need to integrate with existing rule directory scanning in src/services/roo-config/index.ts